### ISSUE TITLE
neuron v2 support

### ISSUE THREAD

**Issue by srid** [fiatjaf/neuron.vim#62]
The query has changed; will need something like this to quickly pull in all zettels. 

```vim
let l:cmd = g:neuron_executable.' -d "'.g:neuron_dir.'" query --cached | jq ".result | .vertices | to_entries[] | [.value]"'
```

**Comment by chiefnoah** [comment]
Sorry, slightly off topic for this issue, but was this actually a v2 change? I've noticed this plugin being broken on neuron `1.9.6.5` for what looks like this issue.

**Comment by srid** [comment]
`1.9.6.5` is pre-v2, and v2 is designed to break a number of things (so as to accomodate the new architectural changes). v1 (which this plugin works with) can still be installed using:

```
nix-env -if https://github.com/srid/neuron/archive/v1.tar.gz
```

**Comment by chiefnoah** [comment]
It looks like this might be a more extensive change than just tweaking the command for listing zettels. Is there any documentation on the return format for `neuron query`? I have some changes necessary to get `refresh_cache_callback` not immediately failing, but it's not immediately clear to me how to fix `zettelQueries` no longer being a direct child of the zettel objects.

I'd also be hesitant to rely on `jq` for this plugin. My WIP changes currently expect that, but IMO it'd be better to just traverse the json structure in viml directly.

**Comment by fiatjaf** [comment]
Will there be a binary release of Neuron 2? I need that to use the plugin against since I deleted all my Nix environment (I love Nix, but my computer is not good enough for it).

**Comment by srid** [comment]
nixpkgs static builds are broken in `master` ... so unless somebody volunteers to resolve it, I'd say unlikely.

**Comment by srid** [comment]
> Is there any documentation on the return format for neuron query? 

Nope; but we should probably clean up its JSON output before finalizing 2.0.

**Comment by chiefnoah** [comment]
> Nope; but we should probably clean up its JSON output before finalizing 2.0.

I sadly don't know Haskell or have time to learn it right now, but once the JSON format is 'stable' for v2.0 I plan on attempting to fix up this plugin for it.

**Comment by fiatjaf** [comment]
> nixpkgs static builds are broken in master ... so unless somebody volunteers to resolve it, I'd say unlikely.

Oh, ok, I didn't know that. I thought binaries were still a thing. I guess I will reinstall nix then.

**Comment by srid** [comment]
I've added `neuron query --zettels` and `neuron query --tags` that should be helpful for editor support. 

Change: https://github.com/srid/neuron/pull/547

Docs: https://neuron.zettel.page/query.html (do a full browser refresh if you see the old docs)

**Comment by srid** [comment]
@chiefnoah take a look at the new (final?) query interface ^

hopefully that's sufficient to use it in neuron.vim

**Comment by srid** [comment]
also checkout `--cached` mode, which really makes things fast on larger zks.

**Comment by chiefnoah** [comment]
The `PluginData` JSON structure is still rather confusing. Can you write up some documentation on how to correctly interpret that?

Example:
```json
    "PluginData": [
      [
        [
          "DirTree",
          []
        ],
        {
          "dirParent": "index",
          "meta": null,
          "childrenTag": null,
          "tags": [
            "root"
          ]
        }
      ],
      [
        [
          "Links",
          []
        ],
        [
          [
            [
              "0de24d86",
              "cf"
            ],
            []
          ],
          [
            [
              "neuron",
              "folge"
            ],
            []
          ]
        ]
      ],
      [
        [
          "Tags",
          []
        ],
        {
          "Queries": [
            [
              "TagQuery_TagZettel",
              "zettelkasten"
            ]
          ],
          "Tagged": [
            "zettelkasten"
          ]
        }
      ],
      [
        [
          "UpTree",
          []
        ],
        []
      ]
    ]
```

The rest of it is fairly obvious, but this is laid out in a rather confusing way and I'm not quite sure how to interpret it.

**Comment by srid** [comment]
PluginData is an internal detail of neuron (used by plugins to store internal data); what are you looking to use it for?
‐‐‐‐‐‐‐ Original Message ‐‐‐‐‐‐‐

**Comment by chiefnoah** [comment]
For fixing `neuron#tags_search`, which queries for zettels based on tags.

**Comment by srid** [comment]
@chiefnoah Just added `neuron query --tag=foo`

**Comment by chiefnoah** [comment]
I think my changes to support v2 are done and can be seen on [this branch](https://github.com/chiefnoah/neuron.vim/tree/neuron-v2).

@fiatjaf how do you want to handle the v1/v2 split for this plugin?
